clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -274,6 +274,12 @@
     private static final String ALLOW_AUTO_CREATE_TOPICS_DOC = "The client-side (consumer) permission to allow auto-topic creation. Both the client-side and the broker-side should enable auto-topic creation in order for a topic to be automatically created";
     public static final boolean DEFAULT_ALLOW_AUTO_CREATE_TOPICS = false;
 
+    public static final String POOL_CLASS_NAME_CONFIG = "linkedin.pool.class.name";
+    public static final String POOL_CLASS_NAME_DOC = "Memory pool class to pool the fetched data from broker. If not specified, uses MemoryPool.NONE.";
```

Comment:
If not specified, i.e. default behavior, we should keep the original behavior (not inc/dec refCnt, no release mem) to be safe since the refCnt approach is only tested in liquid; we want the entire patch to be able to turn on/off with a config to minimize impact to all users; this on/off switch will also make debug/isolate problems easier.